app/code/Magento/CatalogInventory/Model/Quote/Item/QuantityValidator/Initializer/StockItem.php

```diff
@@ -83,22 +83,16 @@ public function initialize(
          */
         if ($quoteItem->getParentItem()) {
             $rowQty = $quoteItem->getParentItem()->getQty() * $qty;
-            /**
-             * we are using 0 because original qty was processed
-             */
-            $qtyForCheck = $this->quoteItemQtyList
-                ->getQty($productId, $quoteItemId, $quoteId, 0);
         } else {
-            $increaseQty = $quoteItem->getQtyToAdd() ? $quoteItem->getQtyToAdd() : $qty;
             $rowQty = $qty;
-            $qtyForCheck = $this->quoteItemQtyList->getQty(
-                $productId,
-                $quoteItemId,
-                $quoteId,
-                $increaseQty
-            );
         }
 
+        /**
+         * we are using 0 because original qty was processed
+         */
+        $qtyForCheck = $this->quoteItemQtyList
```

Comment:
Hi @shikhamis11 @dmytro-ch can you please provide an explanation on how this change is fixing the issue. Also, can you please explain the comment "we are using 0 because original qty was processed" - where the qty was processed?

As far as I remember `\Magento\CatalogInventory\Model\Quote\Item\QuantityValidator\QuoteItemQtyList::getQty` was working incorrectly becouse of `null` is passed as `$quoteItemId ` in certain cases